make-rules/environment.mk

```diff
@@ -48,6 +52,63 @@ component-environment-prep::
 		{ echo "Adding required packages to build environment..."; \
 		  $(PFEXEC) /usr/bin/pkg install --accept -v $(REQUIRED_PACKAGES:%=/%) || [ $$? -eq 4 ] ; }
 
+ZONENAME_PREFIX = bz
+ZONENAME_ID = $(shell echo "$(WS_TOP)" | sha1sum | cut -c0-7)-$(COMPONENT_NAME)
+ZONENAME = $(ZONENAME_PREFIX)-$(ZONENAME_ID)
+
+component-zone-template:
+	$(call separator-line)
+	$(call separator-line,Create template zone)
+	USER_ID=$$(id -u) && \
+	$(PFEXEC) $(ZONE) --prefix $(ZONENAME_PREFIX) create-template -u $${USER} -i $${USER_ID}
+
+component-zone-build:
+	$(call separator-line)
+	$(call separator-line,Create $(ZONENAME))
+	$(PFEXEC) $(ZONE) --prefix $(ZONENAME_PREFIX) spawn-zone --id $(ZONENAME_ID)
+	$(call separator-line,Boot $(ZONENAME))
+	@while $$(true); do \
+		echo "Waiting for zone $(ZONENAME) to boot..."; \
+		$(PFEXEC) /usr/sbin/zlogin -l $${USER} $(ZONENAME) \
+				/bin/true >/dev/null 2>&1 && break; \
+		sleep 10; \
```

Comment:
should we check for zone boot failure or some timeout?

Reply:
I want to keep this light. I will start testing to see, how zones behave about this and if those errors start to appear, i would most like deal with them.